zerver/lib/queue.py

```diff
@@ -203,16 +204,52 @@ def json_drain_queue(self, queue_name: str) -> Iterator[List[Dict[str, Any]]]:
         with self.drain_queue(queue_name) as binary_messages:
             yield list(map(orjson.loads, binary_messages))
 
+    def start_json_consumer(self,
+                            queue_name: str,
+                            callback: Callable[[List[Dict[str, Any]]], None],
+                            batch_size: int=1,
+                            timeout: Optional[int]=None) -> None:
+        if batch_size == 1:
+            timeout = None
+
+        def do_consume(channel: BlockingChannel) -> None:
+            events: List[Dict[str, Any]] = []
+            last_process = time.time()
+            max_processed: Optional[int] = None
+            self.is_consuming = True
+            for method, properties, body in channel.consume(queue_name, inactivity_timeout=timeout):
```

Comment:
To check if I have this right, the intent here with `inactivity_timeout` is that we don't want to block for longer than `sleep_delay` of the queue processor, so we want to get yielded `None` and get to the `callback()` call to process whatever events we have gathered so far? Perhaps we should have a comment here, because it seems obvious now that I typed it out, but initially I puzzled over it for a while